specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/examples/DiskMigrations/Create.json

```diff
@@ -0,0 +1,85 @@
+{
+  "parameters": {
+    "subscriptionId": "04666444-56f0-4d4f-afc5-dbd6b134b084",
+    "location": "redmond",
+    "migrationId": "ec9f79cf-e4d7-4d37-926a-e69c2198be8a",
+    "api-version": "2018-07-30-preview",
+    "targetShare": "\\\\SU1FileServer.mydomain.com\\SU1_ObjStore_1",
+    "disks": [
+      {
+        "id": "/subscriptions/04666444-56f0-4d4f-afc5-dbd6b134b084/providers/Microsoft.Compute.Admin/locations/redmond/disks/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+        "name": "redmond/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+        "type": "Microsoft.Compute.Admin/locations/disks",
+        "location": "redmond",
+        "properties": {
+          "diskId": "423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+          "status": "Unattached",
+          "sharePath": "\\\\SU1FileServer.mydomain.com\\SU1_ObjStore_4",
+          "actualSize": 183,
+          "tenantId": "/subscriptions/4496690e-348e-4dc0-addc-17395c3966ba/resourceGroups/DISKRG6ECD14E6/providers/Microsoft.Compute/Disks/diskrptest9vb55sb15w",
+          "resourceType": "Disk",
+          "diskSku": "Standard_LRS"
+        }
+      },
+      {
+        "id": "/subscriptions/04666444-56f0-4d4f-afc5-dbd6b134b084/providers/Microsoft.Compute.Admin/locations/redmond/disks/48dc9b56-5883-4011-9dc3-0e527f33e6ab",
+        "name": "redmond/48dc9b56-5883-4011-9dc3-0e527f33e6ab",
+        "type": "Microsoft.Compute.Admin/locations/disks",
+        "location": "redmond",
+        "properties": {
+          "diskId": "48dc9b56-5883-4011-9dc3-0e527f33e6ab",
+          "status": "Unattached",
+          "sharePath": "\\\\SU1FileServer.mydomain.com\\SU1_ObjStore_2",
+          "actualSize": 1,
+          "tenantId": "/subscriptions/4496690e-348e-4dc0-addc-17395c3966ba/resourceGroups/DISKRG6ECD14E6/providers/Microsoft.Compute/Snapshots/diskrptestjrlt133h5g8t",
+          "resourceType": "Snapshot",
+          "diskSku": "Standard_LRS"
+        }
+      }
+    ]
+  },
+  "responses": {
```

Comment:
Does PUT and GET not use the same data contract? That's typically not OK. Request should be an action, not a resource. And ideally, migration progress is reflected in migrated objects themselves. See if you can completely eliminate the diskmigrations type and stick with more standard patterns.

Reply:
Right. Migrate could be an action in a resource, and it could be targets at 2 kinds of resources: 
1. shares
Considering DRP doesn't own this kind of resource, we could not take this. But it does look good, if we could make it in the long term.
2. disks 
Based on our PM design, the migration is on disk collection, instead of a single disk. I don't think we have a way to multiple Rest API calls by auto generated SDK and PS. And if we record something in the server side, we will lost track of status if we record in client side.